start.py

```diff
@@ -45,9 +45,11 @@ def save_config_and_exit():
     if len(sys.argv) > 1:
         code_name = sys.argv[1]
     else:
-        code_name = input(
-            "Please provide a short, lowercase code name with no spaces or punctuation for the archive "
-            "you are processing (and make a note of it as you'll need it in future!):\n>> ")
+        code_name = None
+        while (code_name == None or any([x in code_name for x in ["-", "=", ".", " "]])):
```

Comment:
Good idea! Apparently, there's a `string.punctuation` in Python's `string` that would cover any punctuation people might feel like putting in there. It's probably not all dangerous but as this is used to generate filenames, better be safe than sorry. ([looks as if single quotes would need to be added](https://stackoverflow.com/a/18429172), though)

Reply:
A more exhaustive list would be nice, but a whitelist for valid SQL identifiers would be better, I will look if there is some spec with it.

Reply:
The MySQL reference allows the following:

> - Permitted characters in unquoted identifiers:
>   - ASCII: [0-9,a-z,A-Z$_] (basic Latin letters, digits 0-9, dollar, underscore)
>   - Extended: U+0080 .. U+FFFF 
> - Permitted characters in quoted identifiers include the full Unicode Basic Multilingual Plane (BMP), except U+0000: 
>   - ASCII: U+0001 .. U+007F
>   - Extended: U+0080 .. U+FFFF 
> - ASCII NUL (U+0000) and supplementary characters (U+10000 and higher) are not permitted in quoted or unquoted identifiers. 
> - Identifiers may begin with a digit but unless quoted may not consist solely of digits. 
> - Database, table, and column names cannot end with space characters. 

As a person native to a language that includes some wierd letters such as ,,ąęłóćźż", I say we just convert that validation to something like this:

```python
while (code_name == None or any([x.lower() not in code_name for x in "abcdefghijklmnoprstuwxyz"])):
    pass
```
That should be eneugh for our usecase, that is English based.

Reply:
Yes, might as well. The processing is English-based at least, even if the archives aren't, and using nothing but lowercase ASCII will just avoid any kind of weirdness down the line. 👍